src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -249,11 +282,26 @@ private async Task<IEnumerable<PackageReference>> GetPackageReferencesAsync(Pack
                 _lastTimeAssetsModified = fileInfo.LastWriteTimeUtc;
             }
 
-            return packageSpec
+            var installedPackages = packageSpec
                .TargetFrameworks
                .SelectMany(f => GetPackageReferences(f.Dependencies, f.FrameworkName, _installedPackages, assetsPackageSpec, targets))
                .GroupBy(p => p.PackageIdentity)
                .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First());
+
+            // if we're reading the lock file, also update _transitivePackages
+            if (targets != null)
```

Comment:
My thinking here was that if we're already reading the assets file, we should also grab the transitive packages at the same time. This avoids having to read the assets file a second time to get the transitive packages. 
I'll look into how having a single GetPackagesAsync would work.

Reply:
Did you confirm that the implementation actually works like that? My memory of reading the code on Friday was that I believe it will read the assets file twice.

Reply:
The assets file is only read once, whenever a change is made to the file.

Reply:
I changed to to GetAllPackagesAsync so that it's clear that we are getting both direct and transitive packages.